Homework 5

kseniya1manuilova opened the pull request:
Homework#5:
- a new object is created;
- classes are created;
- output of information about the object has been added;
- refactoring.

Homework#6:
- parental class with common properties added;

Homework#7:
- ability to save values to a txt file;
- ability to read values from the file ( at the start);
- ability to add objects via the console added.

Commit:
Homework 5 was added:
- new classes were added
- a new object created
- output of information about the object has been added

Commit:
refactoring added.

Commit:
home task 6 added.

Commit:
home task 7 added:
- ability to save values to a txt file.
- ability to read values from the file ( at the start)

Commit:
As part of the homework, the ability to add objects via the console has been added.

Fulcrum81 commented on src/Main.java:
```diff
@@ -1,94 +1,212 @@
+
+            int selectedOption = scanner.nextInt();
+
+            if (selectedOption == 1) {
```
В таких ситуациях лучше switch/case использовать

kseniya1manuilova replied:
done

Fulcrum81 commented on src/storage/StorageType.java:
```diff
@@ -0,0 +1,21 @@
+package Storage;
+
+public class StorageType {
```
Немного мне непонятна логика именования, но ок...

Fulcrum81 commented on src/Storage/StorageType.java:
```diff
@@ -0,0 +1,21 @@
+package Storage;
+
+public class StorageType {
+    protected String storageType;
```
Класс называется StorageType, "заглавное" поле тоже storageType, небольшая есть нестыковка логическая тут.
Глобально, если мы описываем какую-то сущность, значит в ней должны сожержаться более мелкие ее составляющие.
Например, если у нас класс "Товар", то поле "товар" в нем неуместно, т.к. оно, по сути, является вложенным, нельзя описать какую-то сущность с помощью ее самой, у товара может быть наименование, например, но товар не является составляющей частью товара... Аналогично и здесь, нельзя описать большую сущность StorageType с помощью термина storageType, там должна быть какая-то характеристика StorageType-a. Надеюсь смог изложить, что мне тут не понравилось

kseniya1manuilova replied:
угу, вроде как поняла и поменяла.

Fulcrum81 commented on src/Storage/Bag.java:
```diff
@@ -0,0 +1,74 @@
+package Storage;
```
Пакеты принято именовать со строчной буквы

kseniya1manuilova replied:
done

Fulcrum81 commented on src/Bag.java:
```diff
@@ -0,0 +1,38 @@
+import Storage.Colour;
+
+public class Bag {
```
Не разобрался, зачем Bag-a и Item-a по два?

kseniya1manuilova replied:
забыла удалить после изменения структуры проекта, дан.

Commit:
all comments fixed:
 - class renamed
 - if/else changed to switch/case
 - package renamed
 - duplicated deleted

Commit:
Merge remote-tracking branch 'origin/main' into homework_5

# Conflicts:
#	src/Main.java

kseniya1manuilova committed:
Merge branch 'main' into homework_5
